pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -188,7 +186,9 @@ func (c *AntreaIPAMController) cleanUpStaleIPAddresses() {
 	pods, _ := c.podLister.List(labels.Everything())
 	podsMap := make(map[string]struct{}, len(pods))
 	for _, p := range pods {
-		podsMap[k8s.NamespacedName(p.Namespace, p.Name)] = struct{}{}
+		if p.DeletionTimestamp == nil {
```

Comment:
I don't think I have ever seen this as a condition in our code. However, several of our controller use `k8s.IsPodTerminated`. Can you check if this is what we should be using here?

Reply:
Updated.